src/common/s3_adapter.cpp

```diff
@@ -302,10 +304,11 @@ int S3Adapter::PutObject(const Aws::String &key,
 void S3Adapter::PutObjectAsync(std::shared_ptr<PutObjectAsyncContext> context) {
     Aws::S3::Model::PutObjectRequest request;
     request.SetBucket(bucketName_);
-    request.SetKey(context->key.c_str());
-    request.SetBody(Aws::MakeShared<Aws::StringStream>(
-        "stream",
-        Aws::String(static_cast<char*>(context->buffer), context->bufferSize)));
+    request.SetKey(Aws::String{context->key.c_str(), context->key.size()});
+
+    request.SetBody(Aws::MakeShared<PreallocatedIOStream>(
+        AWS_ALLOCATE_TAG, reinterpret_cast<char *>(context->buffer),
```

Comment:
Can we change buffer type from void* to char * to avoiding reinterpret_cast here?

Reply:
fix